src/main/java/com/exadel/access/control/TrainerControl.java

```diff
@@ -27,46 +28,73 @@ public TrainerControl(UserService userService, TrainingService trainingService,
         this.entryService = entryService;
     }
 
-    public boolean isCoach(long id) {
-        if (userService.getCurrentId()==trainingService.getTrainerId(id))
+    public boolean isTrainerByTrainingId(long id) {
```

Comment:
No, I supposed, that parameter - trainer ID, but it is training ID. So I renamed all methods for clearness.